### ISSUE TITLE
一些代码问题

### ISSUE THREAD

**Issue by PPPPPPPeng** [adversarial-for-goodness/APF#3]
关于您提供的代码，我有如下几个问题想请教您（鉴于您是国内作者我直接用中文了：
1.您在论文中提到的Adversarial Gradient Transfer T网络是如你代码所写，用对抗扰动（对抗噪声）作为输入，得到transferred gradient的吗（g^)，请问g~出现在代码的哪个位置呢，而且我并没有找到公式（4）所示的优化函数的表达式，看到您整个训练过程中只是优化了公式（9）的损失函数。
2. output_type : E  这边E是代表什么意思呢？
3.提高的三种图像变换（rescaling,rotation,color conversion）我没找到在代码中出现的位置，可以麻烦您提醒一下嘛？
4.对于两次用到的ensemble models用于 gradient和ensemble embedding feature exact，我也没找到在代码中出现的位置，可以麻烦您提醒一下嘛？

**Comment by zzhbrr** [comment]
您好，我想请问一下，公式（9）的损失函数优化在哪里呢？还有我想问一下里面的benchmark是什么含义吗，感觉这里面unet的训练损失跟论文中的不大一样。

**Comment by jiamingzhang94** [comment]
Hi @zzhbrr , 第一个问题：PNet_unet.py 中#70 loss_s = tf.reduce_mean(loss_distance_s)。第二个问题，benchmark就是enrolled image。因为tf我也很久不用了，你可以看最新的torch版本，虽然只有g_hat的代码。
